### PR TITLE
Consolidate `Classification` into `TopicalEvent`

### PR DESCRIPTION
**Pull request by brucebolt**
The `Classification` model is currently only inherited by `TopicalEvent`.  Previously, this was shared between `Topic` (known in the UI as 'Policy areas') and `TopicalEvent`, but [the former has previously been removed](https://github.com/alphagov/whitehall/pull/5666).

This split caused confusion when doing work on Topical Events recently and made the code more difficult to work with.

Within this PR, we consolidate `Classification` into `TopicalEvent` and update the associated code.

[Trello card](https://trello.com/c/r2f4CMfF)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

File: app/controllers/admin/topical_event_featurings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+class Admin::TopicalEventFeaturingsController < Admin::BaseController
+  before_action :load_topical_event
```
Comment (BeckaL) [inline app/controllers/admin/topical_event_featurings_controller.rb:2]: So for organisations and world location news, they use 'Feature Lists' and a 'Feature Lists controller', by using `include Featureable`. So just wondering whether we could rationalise this even further by removing this altogether and using the same - I don't think there's a particular reason why classifications should have their own version of this, but maybe I'm missing something. But seems a bit fishy to me.

File: db/migrate/20220726111449_drop_classification_policies_table.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class DropClassificationPoliciesTable < ActiveRecord::Migration[7.0]
```
Comment (JonathanHallam) [inline db/migrate/20220726111449_drop_classification_policies_table.rb:1]: Super minor but there's a link missing in the commit message for this one

File: app/controllers/admin/topical_event_featurings_controller.rb (hunk 1 above)
Comment (BeckaL) [reply app/controllers/admin/topical_event_featurings_controller.rb:2]: Maybe we could do this as a separate PR to this but I think it'd be worth exploring to simplify things, as it's another bit of weirdness in Whitehall that we've encountered

Comment (JonathanHallam) [reply app/controllers/admin/topical_event_featurings_controller.rb:2]: This is a really good idea for a spike or mob. Excellent spot

**Review by JonathanHallam — APPROVED** [review]
Am happy with the code changes, this is a really gnarly bit of work, massive kudos!

**Comment by brucebolt** [comment]
I've cherry picked the `classification_policies` and `classification_relations` table dropping into a separate PR: https://github.com/alphagov/whitehall/pull/6711

**Review by JonathanHallam — APPROVED** [review]
Nice one, thanks for this, super bit of work. Happy now that the more destructive bits of this have been moved elsewhere

**Pull request merged**